### PR TITLE
[MetaSchedule] Generate MetaSchedule Dataset

### PR DESCRIPTION
**Pull request by Kathryn-cat**
In order to build a dataset for improving the cost model for MetaSchedule, I added several files including importing models to TVM, extracting tuning tasks, and sampling measure candidates. Meanwhile, I exposed some methods in C++ to the Python side to assist the process.
[update] In the file `dataset_sample_candidates.py`, I manually save tuning records to json files because `JsonDatabase` is unreasonably slow. I will update it after @junrushao1994 's PR on json parser is merged.

### CODE REVIEW

File: src/meta_schedule/search_strategy/evolutionary_search.cc [hunk 1]
```diff
@@ -719,9 +719,36 @@ class EvolutionarySearch : public SearchStrategy {
                                                     EvolutionarySearchNode);
 };
 
+Array<Schedule> EvolutionarySearchSampleInitPopulation(ObjectRef _self, int num) {
+  const EvolutionarySearchNode* self = _self.as<EvolutionarySearchNode>();
```
Comment (junrushao1994) [inline src/meta_schedule/search_strategy/evolutionary_search.cc:723]: I added the `EvolutionarySearch` class in the latest PR so you may use it directly now

```suggestion
Array<Schedule> EvolutionarySearchSampleInitPopulation(EvolutionarySearch self, int num) {
```

File: src/meta_schedule/search_strategy/evolutionary_search.cc [hunk 2]
```diff
@@ -719,9 +719,36 @@ class EvolutionarySearch : public SearchStrategy {
...
+  std::vector<Schedule> population_vec =
+      std::vector<Schedule>(population.begin(), population.end());
+  std::vector<Schedule> schs = self->state_->EvolveWithCostModel(population_vec, num);
+  for (Schedule sch : schs) {
+    IRModule mod = sch->mod();
+    size_t shash = StructuralHash()(mod);
+    if (!self->state_->measured_workloads_.Has(mod, shash)) {
+      result.push_back(sch);
+    }
```
[the remark this answers is not in the document]
Comment (Kathryn-cat) [reply src/meta_schedule/search_strategy/evolutionary_search.cc:737]: Thanks for the valuable suggestion! Updated :)

File: src/meta_schedule/search_strategy/evolutionary_search.cc (hunk 1 above)
Comment (Kathryn-cat) [reply src/meta_schedule/search_strategy/evolutionary_search.cc:720]: Aha thanks! It's much more convenient.

**Review by junrushao1994 — APPROVED** [review]
Really great work @Kathryn-cat!!!

**Pull request merged**